collectors/ioping.plugin/ioping.plugin.in

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# netdata
+# real-time performance and health monitoring, done right!
+# (C) 2017 Costa Tsaousis <costa@tsaousis.gr>
+# GPL v3+
+#
+# This plugin requires a latest version of ioping.
+# You can compile it from source, by running me with option: install
+
+export PATH="${PATH}:/sbin:/usr/sbin:/usr/local/sbin"
+export LC_ALL=C
+
+if [ "${1}" = "install" ]
+    then
+    [ "${UID}" != 0 ] && echo >&2 "Please run me as root. This will install a single binary file: /usr/local/bin/ioping." && exit 1
+
+    run() {
+        printf >&2 " > "
+        printf >&2 "%q " "${@}"
+        printf >&2 "\n"
+        "${@}" || exit 1
+    }
+
+    download() {
+        local git="$(which git 2>/dev/null || command -v git 2>/dev/null)"
+        [ ! -z "${git}" ] && run git clone "${1}" "${2}" && return 0
+
+        echo >&2 "Cannot find 'git' in this system." && exit 1
+    }
+
+    [ ! -d /usr/src ] && run mkdir -p /usr/src
+    [ ! -d /usr/local/bin ] && run mkdir -p /usr/local/bin
+
+    run cd /usr/src
+
+    if [ -d ioping-netdata ]
+        then
+        run rm -rf ioping-netdata || exit 1
+    fi
+
+    download 'https://github.com/netdata/ioping.git' 'ioping-netdata'
+    [ $? -ne 0 ] && exit 1
+    run cd ioping-netdata || exit 1
+
+    run make clean
+    run make
+    if [ -f /usr/local/bin/ioping ]
```

Comment:
I can use the configured libexec path to run the ioping binary. But I don't see any way to do it for the ioping installation. When a user runs the script manually, we have no information about the latest netdata installation or runtime environment available for use in the script. The definitions from the `.environment` are inaccessible because we don't know where this file is at the moment the user is running the script.

Automatic installation of the ioping at runtime is much worse than forcing the user to do it manually. The same I can say about installing a third party software during a netdata installation process. Most users don't need ioping statistics.

I'd go the way it is now until the ioping maintainers merge a PR which adds netdata support to it. An ioping installation is the process run by the user concsiously and provides him with all the crucial information about it.

Reply:
Hello @vlvkobal, just to confirm: the above comment is before or after our face to face discussions?

Reply:
After the discussion.

Reply:
Moved ioping installation to /usr/libexec/netdata